euclid.py

```diff
@@ -1298,6 +1315,17 @@ def __copy__(self):
 
     copy = __copy__
 
+    def __getitem__(self, key):
+        return (self.w, self.x, self.y, self.z)[key]
+
+    def __setitem__(self, key, value):
+        l = [self.w, self.x, self.y, self.z]
+        l[key] = value
+        self.w, self.x, self.y, self.z = l
+
+    def __iter__(self):
+        return iter((self.w, self.x, self.y, self.z))
+
     def __repr__(self):
         return 'Quaternion(real=%.2f, imag=<%.2f, %.2f, %.2f>)' % \
```

Comment:
artemis repo has `%.4f` everywhere on this line.

Reply:
reverted this to `%.4f`